packages/expo-notifications/ios/EXNotifications/PushToken/PushTokenModule.swift

```diff
@@ -0,0 +1,62 @@
+//  Copyright © 2024 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import UIKit
+import MachO
+
+let onDevicePushTokenEventName = "onDevicePushToken"
+
+public class PushTokenModule: Module {
+  var promiseNotYetResolved: Promise?
+
+  @objc
+  public func onExpoNotificationsRegistrationResult(notification: Notification) {
+    guard let userInfo = notification.userInfo else {
+      return
+    }
+    if let error = userInfo["error"] as? (any Error) {
+      promiseNotYetResolved?.reject(error)
+      promiseNotYetResolved = nil
+    } else if let deviceToken = userInfo["deviceToken"] as? String {
+      promiseNotYetResolved?.resolve(deviceToken)
+      promiseNotYetResolved = nil
+      self.sendEvent(onDevicePushTokenEventName, ["devicePushToken": deviceToken])
+    }
+  }
+
+  public func definition() -> ModuleDefinition {
+    Name("ExpoPushTokenManager")
+
+    Events([onDevicePushTokenEventName])
+
+    OnStartObserving {
```

Comment:
Let's use the new approach for `OnStartObserving`/`OnStopObserving`, where we provide the event name.
```suggestion
    OnStartObserving(onDevicePushTokenEventName) {
```